RIPS/rip-9995.md

```diff
@@ -0,0 +1,199 @@
+---
+rip: 7560
+title: Native Account Abstraction with EOF code sections
```

Comment:
Following our recent discussion, I think this RIP should be rewritten with just one "kind" - the "entrypoints table".  I'm not commenting in specific lines since this change affects a lot of them.  I'll just repeat the basic idea:

- Currently contracts get executed from offset 0, the default entrypoint.
- Let's keep it a single bytecode (no sections) but use EOF to add a table of entrypoints.  The default remains 0.
- Each entry in this table is (type,offset).  E.g. (AA_SENDER_VALIDATION, 0x58), or (AA_PAYMASTER_VALIDATION, 0x312).  There's always the implicit (DEFAULT, 0) but no need to waste space on it unless we think it'll be useful to have a contract that doesn't have the normal code.
- TransactionType4 is only valid if its sender has an AA_SENDER_VALIDATION entrypoint, and the validation frame starts executing from that offset.
- TransactionType4 with a nonzero paymaster is only valid if its paymaster has an AA_PAYMASTER_VALIDATION entrypoint, and the paymaster validation frame starts executing from that offset.
- It removes the need for MAGIC return values, ABI, etc.
- This will be easy to extend in the future for non-AA use cases.
- Compilers may add a pragma to mark functions as entrypoints, e.g. `#pragma AA_SENDER_VALIDATION` or add a keyword, similar to `constructor` and `receive`. These functions may or may not be marked `external`, indicating whether the function should also be accessible via the dispatch code in offset 0.  All of these details are outside the scope of the RIP and should remain at the language level.